src/Umbraco.Core/PublishedCache/IMediaCacheService.cs

```diff
@@ -26,4 +26,5 @@ public interface IMediaCacheService
     Task SeedAsync(CancellationToken cancellationToken);
 
     void Rebuild(IReadOnlyCollection<int> contentTypeIds);
+    IEnumerable<IPublishedContent> GetByContentType(IPublishedContentType contentType);
```

Comment:
Shouldn't this be async (either `Task<IEnumerable<IPublishedContent>>` or `IAsyncEnumerable<IPublishedContent>`) and use the content type ID/key instead (similar to `RebuildMemoryCacheByContentTypeAsync()`?

```suggestion

    IAsyncEnumerable<IPublishedContent> GetByContentTypeAsync(Guid contentTypeKey);
```

Reply:
This is only implemented to avoid breaking changes 😁 

It is obsolete on the `IPublishedCache` interface: 
![image](https://github.com/user-attachments/assets/0c8cf4cd-9d52-47dd-95ff-71e5cb91170f)